tests/test_tools.py

```diff
@@ -250,3 +253,61 @@ def get_time() -> str:
             "destructiveHint": None,
             "openWorldHint": None,
         }
+
+
+async def test_convert_langchain_tool_to_fastmcp_tool():
+    @tool
+    def add(a: int, b: int) -> int:
+        """Add two numbers"""
+        return a + b
+
+    class AddInput(BaseModel):
+        """Add two numbers"""
+
+        a: int
+        b: int
+
+    @tool("add", args_schema=AddInput)
+    def add_with_schema(a: int, b: int) -> int:
+        return a + b
+
+    class AddTool(BaseTool):
+        name: str = "add"
+        description: str = "Add two numbers"
+        args_schema: ArgsSchema | None = AddInput
+
+        def _run(self, a: int, b: int, run_manager: CallbackManagerForToolRun | None = None) -> int:
+            """Use the tool."""
+            return a + b
+
+        async def _arun(
+            self, a: int, b: int, run_manager: CallbackManagerForToolRun | None = None
+        ) -> int:
+            """Use the tool."""
+            return self._run(a, b, run_manager=run_manager)
+
+    for tool_ in [add, add_with_schema, AddTool()]:
```

Comment:
pytest parameterize maybe?  if we can put the tools in the global namespace?

The main goal is to support the failing case, so it's easy to identify which of the test cases failed.